hazelcast/src/main/java/com/hazelcast/sql/impl/QueryUtils.java

```diff
@@ -190,6 +193,17 @@ public static SqlColumnMetadata getColumnMetadata(String columnName, QueryDataTy
         for (Partition part : parts) {
             Member owner = part.getOwner();
 
+            if (owner == null) {
```

Comment:
I cannot explain the exact situations when the partition could be unassigned. Some of them - member failure, FROZEN cluster state. The main thing is that the `null` is in the contract of this method (see JavaDoc):
> It can be that null is returned if the owner of a partition has not been established

Reply:
The partition table is also calculated lazily so if you create a new cluster and don't call any partition operations, the partition table is not established yet and owners are null.

Reply:
Does it mean that one member failure would trigger this exception for any SQL query?

Reply:
Member failure may trigger temporal errors until the partition table is recalculated.